### ISSUE TITLE
`async` as property name is broken in jspm 0.17.0-beta.47

### ISSUE THREAD

**Issue by laurentgoudet** [jspm/jspm-cli#2405]
Using `async` as a property name (e.g. as used in `jQuery.ajax()` calls for instance) prevents `jspm` to be upgraded to a version above `0.17.0-beta.43`, as https://github.com/systemjs/builder/commit/ab71fc5d3d20d1c1e123af3ae273bf1af8aa7911 in SystemJS Builder 0.16.10, by turning on ES2017 mode, triggers an Acorn bug with `async` as property name fixed in https://github.com/acornjs/acorn/releases/tag/4.0.3.

In short, this is broken in Acorn 4.0.1, used by Rollup 0.36.4 (used by SystemJS Builder):

```
> file = fs.readFileSync('./index.js').toString()
'export const testObj = {\n  async: true\n};\n'
> acorn.parse(file, { ecmaVersion: 8, allowReserved: true, sourceType: 'module' })
SyntaxError: Unexpected token (2:7)
```

And it works in 4.0.3:

```
> file = fs.readFileSync('./index.js').toString()
'export const testObj = {\n  async: true\n};\n'
> acorn.parse(file, { ecmaVersion: 8, allowReserved: true, sourceType: 'module' })
Node {
  type: 'Program',
  start: 0,
  end: 42,
  body:
   [ Node {
       type: 'ExportNamedDeclaration',
       start: 0,
       end: 41,
       declaration: [Object],
       specifiers: [],
       source: null } ],
  sourceType: 'module' }
```

In addition, although not related, `jspm` < `0.17.0-beta.43` blows up in Safari 11.1 (current stable) due to a buggy Link Preload implementation (apparently fixed in Safari Technology Preview), triggered by the valid-yet-buggy SystemJS behavior fixed in https://github.com/systemjs/systemjs/commit/be6d7a6dd6d5ee99a1ebb154e38b7d23987d59a2 (in short Safari HTTP cache gets corrupted when an in-progress Link preload tag is removed from the DOM -> https://jsfiddle.net/qhv45nqh/3/). This means people using `depCache` must upgrade to jspm >= `0.17.0-beta.43` in order to support Safari 11.1.

![screen shot 2018-04-25 at 14 30 43](https://user-images.githubusercontent.com/2169585/39420211-d3b0c364-4c63-11e8-8451-f9bf5f4b421f.png)

**Comment by guybedford** [comment]
Thanks for the updates here.

So to summarize what is needed:

* SystemJS builder 0.16 should update Rollup
* SystemJS 0.19 should implement the preload fix from https://github.com/systemjs/systemjs/commit/be6d7a6dd6d5ee99a1ebb154e38b7d23987d59a2

Both sound fairly straightforward I think.

If you're able to PR that would be a great help, otherwise I will aim to look into these soon.

**Comment by laurentgoudet** [comment]
> SystemJS builder 0.16 should update Rollup

That's correct, I can confirm https://github.com/systemjs/builder/pull/847 fixes it (I have a small repro repo at https://github.com/laurentgoudet/rollup-bug).

> SystemJS 0.19 should implement the preload fix from systemjs/systemjs@be6d7a6

The preload bug had been introduced in 0.20.0-rc.8 (https://github.com/systemjs/systemjs/commit/bb62e580ddb05e429384308d664e7661e19b7343) so SystemJS 0.19 / JSPM 0.16 should already be fine.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.48.
